awssql/utils/sliding_expiration_cache.go

```diff
@@ -129,6 +128,19 @@ func (c *SlidingExpirationCache[T]) Remove(key string) {
 	}
 }
 
+func (c *SlidingExpirationCache[T]) cleanupIfExpired(key string) {
+	c.lock.Lock()
```

Comment:
Does it make sense to just use `defer c.lock.Unlock()` here for simplicity?

Reply:
The intent here was to release the lock as soon as possible. So in the case the cache item is found, we release locks before running the disposal function. 
Unfortunately, the side effect of that is that we cant defer releasing the lock.